financial/bill_splitter.py

```diff
@@ -0,0 +1,109 @@
+
+from typing_extensions import TypedDict
+
+
+class Contribution(TypedDict):
+    name: str
+    contri: int
+
+
+class Solution(TypedDict):
+    name: str
+    payment: list[Contribution]
+
+
+class Result(TypedDict):
+    sol: list[Solution]
+    total: int
+    each: float
+    pool: list[Contribution]
+
+
+# function to split bill among group
+# pool: [...{'name':str,'contri':int},{'name':str,'contri':int}]
+def split_bill(pool: list[Contribution]) -> Result:
+    """
+    >>> split_bill([{'name': 'sam', 'contri': 500}, {'name': 'rohan', 'contri': 200}, {'name': 'john', 'contri': 50}])
+    {'sol': [{'name': 'sam', 'payment': [{'name': 'rohan', 'contri': 50.0}, {'name': 'john', 'contri': 200.0}]}], 'total': 750, 'each': 250.0, 'pool': [{'name': 'sam', 'contri': 500}, {'name': 'rohan', 'contri': 200}, {'name': 'john', 'contri': 50}]}
+    """
+    contri_list = [x["contri"] for x in pool]
+    total = sum(contri_list)
+    each = total / len(contri_list)
+    more = []
+    less = []
+    sol = []
+    for i in pool:
+        if i["contri"] < each:
+            less.append({"name": i["name"], "contri": each - i["contri"]})
+        else:
+            more.append({"name": i["name"], "contri": i["contri"] - each})
+    for i in more:
+        a = i["contri"]
+        m = [{"name": k["name"], "contri": 0} for k in less]
+        for j in range(len(less)):
+            b = less[j]["contri"]
+            if a == 0:
+                m[j]["contri"] = 0
+            elif a - b == 0:
+                a = a - b
+                m[j]["contri"] = b
+                less[j]["contri"] = 0
+            elif a - b > 0:
+                a = a - b
+                less[j]["contri"] = 0
+                m[j]["contri"] = b
+            elif a - b < 0:
+                less[j]["contri"] = b - a
+                m[j]["contri"] = a
+                a = 0
+        sol.append({"name": i["name"], "payment": m})
+    return {"sol": sol, "total": total, "each": each, "pool": pool}
+
+
+# function to print solution in a format
+# result is the value returned by splitBill(pool) function
+def print_solution(result: Result) -> None:
+    """
+    >>> print_solution({'sol': [{'name': 'sam', 'payment': [{'name': 'rohan', 'contri': 50.0}, {'name': 'john', 'contri': 200.0}]}], 'total': 750, 'each': 250.0, 'pool': [{'name': 'sam', 'contri': 500}, {'name': 'rohan', 'contri': 200}, {'name': 'john', 'contri': 50}]})
+    sam  paid    $ 500
+    rohan  paid    $ 200
+    john  paid    $ 50
+    -------------------------------------------------------------------------------
+    Total pool amount  : $ 750
+    Per head           : $ 250.0
+    -------------------------------------------------------------------------------
+    rohan  should pay $ 50.0  to  sam
+    john  should pay $ 200.0  to  sam
+    -------------------------------------------------------------------------------
+    """
+    for i in result["pool"]:
+        print(i["name"], " paid    $", i["contri"])
```

Comment:
Could you use an f-string for this instead?

Reply:
sure. i will update the pr